packages/aggregator/src/__tests__/analytics/periodFromAnalytics.test.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { periodFromAnalytics } from '../../analytics/periodFromAnalytics';
+
+describe('periodFromAnalytics()', () => {
```

Comment:
Same here, I believe that the non-parameterized version will be easier to read... Sorry for the back and forth!

I guess it's a bit of a judgement call in those cases. I was also probably over emphatic about parameterized tests early on in this project, in practice it seems they are a useful tool but not a "silver bullet". I appreciate that you took the time to go this way though!

The ideal use case would be when using simple input/output combinations. If you are in doubt, I suggest converting one test block to parameterized and see how it looks like. It may take a bit of extra time experimenting but I think it's worth it in the long run!

Reply:
Don't feel sorry Kostas. Thanks for pointing it out. This was done before you mentioned compromising and I forgot to make it right. I should check all code again later. Anyway, hopefully now I 'extend' your judgement mostly :sunglasses:

Reply:
Thanks for that, also keen to listen to your judgement on this matter! Feel free to take decisions based on our taste in this subject for the following work. If it doesn't feel right, we don't have to use parameterized tests 🙌